torch/distributed/_tensor/dispatch.py

```diff
@@ -198,11 +196,22 @@ def default_tensor(spec: DTensorSpec) -> torch.Tensor:
         # communicate the result to all ranks for some operators that return scalar value
         if output_sharding.output_spec is None:
             if op_call == aten.equal.default:
-                obj_list = [None for _ in range(dist.get_world_size())]
-                dist.all_gather_object(obj_list, local_results)  # type: ignore[possibly-undefined]
-                obj_list = list(filter(lambda x: x is not None, obj_list))
-                # perform reduce on the collection with AND op
-                local_results = functools.reduce(operator.and_, obj_list, True)
+                res_tensor_list = [
```

Comment:
we should not use `dist.all_gather`, instead we should use
```
out = funcol.all_gather_tensor(torch.tensor(local_res, dtype=torch.bool, device=mesh.device_type))

out = torch.all(out).item()
```

Reply:
Right. Thx for pointing out! I had misunderstanding here: it's not object collectives that are not traceable but `c10d` collectives.